homeassistant/helpers/entity.py

```diff
@@ -246,7 +246,7 @@ class Entity(ABC):
     _attr_device_info: DeviceInfo | None = None
     _attr_entity_picture: str | None = None
     _attr_entity_registry_enabled_default: bool
-    _attr_extra_state_attributes: MutableMapping[str, Any]
+    _attr_extra_state_attributes: Mapping[str, Any]
```

Comment:
It's allowed to mutate it if entity manages it and it's non shared. Not sure if this type allows that.

Reply:
Unfortunately this need to be `Mapping` to be able to assign `MappingProxyType` to it. Entities can choose to overwrite the type with `MutableMapping` if they what to modify it. I needed to do that in several places here, e.g., for `airvisual`.

The other way round (Entity -> `MutableMapping` and integration -> `Mapping`) doesn't work since `Mapping` only supports a subset of `MutableMapping` and is thus incompatible.

Reply:
Maybe we should just make `attribution` an entity property instead?

It's not really in line with our general design to have static info be part of state attributes. It feels wrong to go out of our way to allow integrations to define the extra state attributes to be static and optimize for that.